fff

```diff
@@ -171,6 +175,37 @@ read_dir() {
     cur_list=("${list[@]}")
 }
 
+# Convert number to a human readable format (with K, M, G suffixes)
+human_readable()
+{
+    local s=$1
+    if [[ $s -lt 1024 ]]; then
+        echo "$s\\e[33m"
+    elif [[ $s -lt 1048576 ]]; then
+        echo "$((s / 1024))\\e[32mK"
+    elif [[ $s -lt $((1048576 * 1024)) ]]; then
+        echo "$((s / 1048576))\\e[32mM"
+    elif [[ $s -lt $((1048576 * 1048576)) ]]; then
+        echo "$((s / (1048576 * 1024)))\\e[32mG"
+    else
+        echo "$s"
+    fi
+}
+
+# Show file size in a human readable format, only for a regular files
+human_readable_size()
```

Comment:
Functions shouldn't `echo` or `printf`. They should modify a global to avoid subshells.